addon/components/fl-input.hbs

```diff
@@ -1,28 +1,31 @@
-<div class="floating-label {{@containerClass}}
-    {{~if @inset " fl-inset-placeholder"}}
-    {{~if this.hasFocus " fl-has-focus"}}
-    {{~if this.hasContent " fl-has-content"}}
-    {{~if this.hasFocusOrContent " fl-has-focus-or-content"}}"
+<div
+    class="floating-label
+        {{@containerClass}}
+        {{if @inset " fl-inset-placeholder"}}
+        {{if this.hasFocus " fl-has-focus"}}
+        {{if this.hasContent " fl-has-content"}}
+        {{if this.hasFocusOrContent " fl-has-focus-or-content"}}"
     {{on "focusin" this.onFocusIn}}
     {{on "focusout" this.onFocusOut}}
 >
     {{#if hasBlock}}
         {{yield}}
         {{#if @placeholder}}
-            <label class="control-label fl-placeholder-label">
+            <label class="control-label fl-placeholder-label" for="{{this.id}}">
                 {{@placeholder}}
             </label>
         {{/if}}
     {{else}}
         <Input
             @value={{@value}}
             class="{{this.inputBaseClass}}"
+            id={{this.id}}
             ...attributes
         />
         {{#if @placeholder}}
-            <label class="control-label fl-placeholder-label">
+            <label class="control-label fl-placeholder-label" for="{{this.id}}">
```

Comment:
we have a mix of wrapping some curly-only bound attribute values in double quotes `for="{{this.id}}"`, and some without `id={{this.id}}`. 

For the code style rule, I personally prefer not wrapping in quotes if you're only passing it a bound value, as its unnecessary and adds noise. And only using quotes if passing in a literal string (`for="foo"`), or if doing inline concatenation of bound and literal values (`for="foo-{{this.bar}}"`)